### PR TITLE
Repo knowledgeable bump formula pr

### PR DESCRIPTION
**Pull request by morganrconnolly**
- [ ] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [ ] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/test/PATH_spec.rb).
- [ ] Have you successfully run `brew style` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?

-----

### CODE REVIEW

**Comment by issyl0** [comment]
Thanks for starting on this PR, @morganrconnolly. I’ll take a look at it this weekend!

**Comment by issyl0** [comment]
Thanks for starting on this!

I found your [comment on the original issue](https://github.com/Homebrew/brew/issues/6341#issuecomment-538681350) and I'll answer it here. I think it's a good idea to get the commands running locally for you so you can test your changes, as I've noticed some syntax issues.

> I'm not sure what the appropriate development workflow is for brew commands, e.g. how do modify a command, rebuild brew and then test the modified command. Is there any documentation about this you can point me towards?

There are docs on [where to run commands from and how to set up your fork](https://docs.brew.sh/How-To-Open-a-Homebrew-Pull-Request#core-brew-code-related-pull-request). You can then checkout your existing forked branch and run the code you're editing (`brew bump-formula-pr`) to test your changes - there are plenty of out of date formula to try running that command with - [you can look at the open formula PRs on Homebrew/homebrew-core](https://github.com/Homebrew/homebrew-core/pulls) for inspiration.

Do you run Homebrew on macOS or Linux? If macOS, you'll probably want to test the functionality in our Linux-based Docker container via `docker pull homebrew/brew` and the above steps.

If you get stuck, or need more advice, do comment here and we can work through it. I'm happy to help! :smile:

File: Library/Homebrew/dev-cmd/bump-formula-pr.rb [hunk 1]
```diff
@@ -297,7 +297,19 @@ def bump_formula_pr
         ohai "git checkout --quiet -"
         ohai "create pull request with GitHub API"
       else
-
+        #determine if this formula has tap  linuxbrew-core
+        if formula.tap.full_name.includes? "linuxbrew-core"
+          #try to identify formula in homebrew core
+            begin 
+              linuxbrew_core_formula = Formulary.factory(formula.gsub('linuxbrew-core','homebrew-core'))
```
Comment (MikeMcQuaid) [inline Library/Homebrew/dev-cmd/bump-formula-pr.rb]: `formula.full_name.gsub`

File: Library/Homebrew/dev-cmd/bump-formula-pr.rb [hunk 2]
```diff
@@ -297,7 +297,19 @@ def bump_formula_pr
...
+            #if there's ambiguity print an informative message
+            rescue TapFormulaAmbiguityError 
+              odie "Unable to identify which homebrew-core formula corresponds to this linuxbrew formula. Please open PR against homebrew-core for corresponding formula"
```
Comment (MikeMcQuaid) [inline Library/Homebrew/dev-cmd/bump-formula-pr.rb]: When does this occur?

File: Library/Homebrew/dev-cmd/bump-formula-pr.rb [hunk 3]
```diff
@@ -297,7 +297,19 @@ def bump_formula_pr
         ohai "git checkout --quiet -"
         ohai "create pull request with GitHub API"
       else
-
+        #determine if this formula has tap  linuxbrew-core
+        if formula.tap.full_name.includes? "linuxbrew-core"
```
Comment (MikeMcQuaid) [inline Library/Homebrew/dev-cmd/bump-formula-pr.rb]: `if OS.linux? && formula.tap.core_tap?`

File: Library/Homebrew/dev-cmd/bump-formula-pr.rb (hunk 2 above)
Comment (dawidd6) [reply Library/Homebrew/dev-cmd/bump-formula-pr.rb]: Linux specific formulae I would say.

**Comment by dawidd6** [comment]
After thinking about it, I would say we need to:
1. Fetch `homebrew-core`
2. Check if formula exists in `homebrew-core` (is not a linux-specific one)
  3. Make a new branch from `homebrew-core` master
  4. Replace the formula tap name from `linuxbrew-core` to `homebrew-core`
5. Proceed

**Comment by dawidd6** [comment]
Alright, I have somewhat working solution:
https://github.com/dawidd6/brew/tree/bumpable-linux

Here is the PR created using `brew bump-formula-pr` from above:
https://github.com/Homebrew/homebrew-core/pull/46424

It didn't check open PRs properly tho as a PR for this already existed:
https://github.com/Homebrew/homebrew-core/pull/46197

**Pull request closed without merging**